pkg/server/cluster_server.go

```diff
@@ -48,13 +54,78 @@ func NewClusterServer(kubeConfig, apiserverURL string) (Server, error) {
 		return nil, fmt.Errorf("Failed to create Kubernetes rest client: %v", err)
 	}
 
+	kc, err := clientset.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	mc := v1.NewForConfigOrDie(restConfig)
+	oc := oconfigv1.NewForConfigOrDie(restConfig)
 	return &clusterServer{
+		kubeClient:     kc,
+		configClient:   *oc,
 		machineClient:  mc,
 		kubeconfigFunc: func() ([]byte, []byte, error) { return kubeconfigFromSecret(bootstrapTokenDir, apiserverURL) },
 	}, nil
 }
 
+// remoteAddrAlreadyIsNode returns true if the given remoteAddr is assigned
+// to a provisioned node.
+func remoteAddrAlreadyIsNode(kc clientset.Interface, remoteAddr string) (string, error) {
+	nodes, err := kc.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, node := range nodes.Items {
+		// Ignore NodeReady=false nodes; this allows reprovisioning them.
```

Comment:
Node not ready doesn't mean node is getting re-provisioned. where did we get that assumption?